test/Microsoft.NET.Build.Tests/GivenThatWeWantToPreserveCompilationContextForBuild.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToPreserveCompilationContextForBuild : SdkTest
+    {
+        public GivenThatWeWantToPreserveCompilationContextForBuild(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [WindowsOnlyFact]
+        public void It_supports_copylocal_false_references()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "CopyLocalFalseReferences",
+                TargetFrameworks = "net461",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties.Add("PreserveCompilationContext", "true");
+            testProject.PackageReferences.Add(new TestPackageReference("NETStandard.Library.NETFramework", "2.0.0-preview2-25330-01", null));
```

Comment:
I thought of that approach, but decided against it because it was "too unit-testy" and this felt more like what a user is actually going to do.

I think when https://github.com/dotnet/sdk/issues/1267 is implemented, this test should be updated accordingly to what the new officially supported approach is for users to use netstandard2.0 on their desktop apps.